packages/picker/src/Picker.ts

```diff
@@ -253,26 +248,20 @@ export class PickerBase extends SizedMixin(Focusable) {
         if (
             !this.popover ||
             !this.optionsMenu ||
-            this.optionsMenu.children.length === 0
+            !this.reparentableChildren ||
+            this.reparentableChildren.length === 0
         ) {
             this.menuStateResolver();
             return;
         }
 
-        this.placeholder = document.createComment(
-            'placeholder for optionsMenu'
+        this.elementReparenter = new ElementReparenter(
```

Comment:
Being you've chosen to make this an object oriented API would it make better sense to recycle the reparenter rather than newing it every time?

I could alternatively see it as a functional API, a la:
```
import { reparentChildren } from 'shared';

this.reclaimChildren = reparentChildren(children, target); 

if (this.reclaimElements) {
    this.reclaimChildren();
    this.reclaimChildren = undefined;
}
```
I know right, a web component admiring suggesting a functional API... 😏

Reply:
In the realm of a class, you will find many parallels with reactive controller approach that the Polymer team is formalizing for the upcoming release of lit-element: https://github.com/Polymer/lit-html/blob/lit-next/packages/reactive-element/CHANGELOG.md

Reply:
The functional approach definitely seems better here, and I even tried that first. I was passing back state and not a callback though which I didn't like.